### PR TITLE
Feature/basic UI

### PR DESCRIPTION
**Pull request by ulisesedu7**
## Description
This pull request creates the main app interface for the user to use the school library

**Task:** https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_ui.md

This PR makes the following changes:
- Creates main and app files
- Initialize all the app features to make it executable in the command line
- Modify some of the previous methods to make everything works

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix, feature, or code refactoring that would cause existing functionality not to work as expected)
- [X] Test (change related to testing)

## Checklist

- [X] My code follows the style guidelines of this project (no linters errors)
- [X] I have checked my code and corrected any misspellings
- [X] I have commented on my code
- [X] I have deleted any unused code as comments in my codebase

### CODE REVIEW

File: main.rb [hunk 1]
```diff
@@ -1,11 +1,121 @@
...
+    person_i = gets.chomp.to_i
+    puts ''
+
+    # Date option
+    puts 'Date: '
+    date = gets.chomp
+
+    app.create_rental(date, person_i, book_i)
+  end
+end
```
Comment (CollinsTatang) [inline main.rb]: - [x] Please fix this error instead to create rental a the list of books is the display 
rental

File: student.rb [hunk 2]
```diff
@@ -2,13 +2,14 @@
...
-  end
-
   # Belong-to relationship
   attr_reader :classroom
+  attr_accessor :parent_permission
+
+  def initialize(age, name, classroom = 'none', parent_permission: true)
+    super(age, name, parent_permission)
+    @classroom = classroom
+  end
```
Comment (CollinsTatang) [inline student.rb]: - [x] Please fix this error generated  when a user wants to create a new student
stude

**Review by CollinsTatang — CHANGES_REQUESTED** [review]
Hi @ulisesedu7, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

### **Highlights 🚀 :heavy_check_mark: :heavy_check_mark:** 

- The repo looks professional.:heavy_check_mark:
- Git/GitHub flow best practices have been followed:heavy_check_mark:
- There are no linter errors:heavy_check_mark:
- You have descriptive PR:heavy_check_mark:

## Required changes 
It's ok to have bugs in your code. Please, see this code review as a chance for you to learn practical coding skills and learn new tips that can improve your code quality and help you get a better job.
_Check the comments under the review._
## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @CollinsTatang in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ulisesedu7** [comment]
@CollinsTatang Thank you very much, I'll start working on the problem!

**Review by zainsadaqat — APPROVED** [review]
# APPROVED!!!! 🎉 🎉 🎉 🎉 🎉 🎉
## Hey @ulisesedu7  😊 😃 

#### Great Job 😄 👏 You've done it. Your project is complete!  There is nothing else to say other than... it's time to merge it 👍🏻 ✅ 

# Congratulations!!! 🎉 🥳 👏 🙌 🎊 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ulisesedu7** [comment]
@zainsadaqat Thank you very much!

**Pull request merged**